### PR TITLE
Prevent interspecies breeding

### LINKED ISSUE #2957: Interspecies breeding is possible
e.g: When I feed one sheep and one cow a little cow or sheep is born.
Cow with sheep might be somehow reasonable. But pig with chicken, that is not very natural.

### PR DESCRIPTION
**Pull request by LogicParrot**
Fixes #2957.

### CODE REVIEW

**Comment by LogicParrot** [comment]
Testcase:
Feed grain to a chicken and a cow. Did they breed?

File: src/Mobs/PassiveMonster.cpp [hunk 1]
```diff
@@ -158,15 +158,19 @@ void cPassiveMonster::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
...
 				{
 					// if we're the same species as someone around and they don't have a partner, start mating with them
-					if ((a_Entity->GetEntityType() == m_Me->GetEntityType()) && (a_Entity != m_Me))
+					if ((a_Entity->GetEntityType() == etMonster) && (a_Entity != m_Me))
 					{
 						cPassiveMonster * Me = static_cast<cPassiveMonster*>(m_Me);
 						cPassiveMonster * Partner = static_cast<cPassiveMonster*>(a_Entity);
-						if (Partner->IsInLove() && (Partner->GetPartner() == nullptr))
+
+						if (Partner->GetMobType() == Me->GetMobType())
```
Comment (madmaxoft) [inline src/Mobs/PassiveMonster.cpp]: How about a fast bailout here? negative condition, `return`.

Comment (LogicParrot) [inline src/Mobs/PassiveMonster.cpp]: Okay. This should eliminate all nesting inside `Item`.

Comment (LogicParrot) [inline src/Mobs/PassiveMonster.cpp]: Done.

**Comment by madmaxoft** [comment]
Nice, this is an exemplary code. Well written and well commented :)

**Comment by LogicParrot** [comment]
Thank you :)

**Comment by LogicParrot** [comment]
Tested. Works.

**Comment by SphinxC0re** [comment]
Rebase and squash. LGTM so I think this can be merged afterwards

**Comment by LogicParrot** [comment]
Squashed and rebased. Merge if CI succeeds.

**Pull request merged**